### ISSUE TITLE
Sortierungsproblem 4

### ISSUE THREAD

**Issue by IxTheoKm** [ubtue/ixTheo#434]
Sonderzeichen werden mit sortiert. In alphabetisch nach Titeln sortierten Treffermengen stehen sie dadurch ganz am Ende des Alphabets:
Bsp.:
https://ixtheo-test.de/Author/Home?author=Benedikt+XVI.+Papst+1927-&type=Author&sort=title&page=125
https://ixtheo-test.de/Author/Home?author=Dierken%2C+J%C3%B6rg+1959-&type=Author&sort=title&page=3

**Comment by mtrojan-ub** [comment]
![grafik](https://cloud.githubusercontent.com/assets/26873381/25805129/d1dd3ed8-33fe-11e7-810f-705d2822a843.png)

**›**Religion**‹** als Thema Evangelischer Theologie

**Comment by mtrojan-ub** [comment]
Um das Problem zu lösen benötigen wir ggf. ebenfalls andere Feldtypen.

Die Solr-Anleitung gibt einen Hinweis auf ein Solr-Plugin, das wir evtl verwenden können:
https://lucene.apache.org/solr/guide/6_6/language-analysis.html#LanguageAnalysis-UnicodeCollation

Ich werde mal testen ob sich das für unsere Zwecke eignet und wir damit die Sortierprobleme lösen können.

**Comment by mtrojan-ub** [comment]
Habe hierzu einen Issue für solrmarc geöffnet.

siehe https://github.com/solrmarc/solrmarc/issues/79

**Comment by mtrojan-ub** [comment]
[siehe ptah](https://ptah.ub.uni-tuebingen.de/Author/Home?author=Dierken%2C+J%C3%B6rg+1959-&type=Author&sort=title&page=3)

![grafik](https://user-images.githubusercontent.com/26873381/34937056-bae957c4-f9e3-11e7-9fb6-938bda8069ad.png)

Den Datensatz "Religion als Thema evangelischer Theologie" gibts tatsächlich zweimal, mit unterschiedlichen ISSNs. Der Datensatz mit "›Religion‹" steht nun aber nicht mehr ganz am Ende der Sortierung.

(Ich habe nun doch unser Mixin erweitert. Es verwendet die Standard solrmarc Funktion und berücksichtigt zudem diverse Unicode-Sonderzeichen für die Sortierung)

**Comment by IxTheoKm** [comment]
Auf PTAH jetzt bis auf eine Ausnahme o.k.:
ein Titel mit einleitendem spanischen Fragezeichen wird noch ganz am Ende angezeigt:
https://ptah.ub.uni-tuebingen.de/Author/Home?sort=title&author=%22%5C%22%5C%5C%22Benedikt+XVI.+Papst+1927-%5C%5C%22%5C%22%22&type=Author&page=137

Ich habe in den Titeldaten ein Sortierzeichen erfasst, um das zu umgehen. Allerdings sollte das Satzzeichen bei der Sortierung generell übergangen werden. Ich weiß nicht, ob es evtl. noch weitere Zeichen geben kann, die weiterhin mit sortiert werden.

**Comment by mtrojan-ub** [comment]
Soll das Unicode Zeichen "…" ebenfalls entfernt werden? siehe [ptah](https://ptah.ub.uni-tuebingen.de/Search/Results?sort=title&page=89505&type=AllFields)

![grafik](https://user-images.githubusercontent.com/26873381/34951106-77c67094-fa15-11e7-8007-8abae3f65357.png)

**Comment by IxTheoKm** [comment]
Ja, unbedingt. Normalerweise wird bisher ein Sortierzeichen vor das erste zu ordnende Wort gesetzt. Bei Verlagsdateneinspielungen fehlt dies aber regelmäßig. Alle wie auch immer gestalteten Anführungszeichen und die Punkte müssen bei der Sortierung übergangen werden. (Um die korrekte Umschrift von "Daimon" kümmere ich mich.)

**Comment by IxTheoKm** [comment]
Lassen sich auch überzählige Leerzeichen übergehen:
https://ptah.ub.uni-tuebingen.de/Search/Results?sort=title&page=89503&type=AllFields
![sortierung](https://user-images.githubusercontent.com/17004223/34952542-65e71b44-fa1a-11e7-9dfb-c0c3a3a5244e.png)
Vor " Entre" und " Protestantisme" sind überzählige Leerzeichen erfasst, die offensichtlich auch sortiert werden. Ich kann die Daten korrigieren, aber es kommt sicher immer wieder mal vor.

**Comment by mtrojan-ub** [comment]
In solr selbst kann ich keine überzähligen Leerzeichen finden, weder bei Entre noch bei Protestantisme. Haben Sie die Datensätze ggf bereits korrigiert? Können Sie mir noch ein weiteres Beispiel nennen?

![grafik](https://user-images.githubusercontent.com/26873381/34977920-c85e64c6-fa9c-11e7-911c-e45c9d5009b7.png)

**Comment by IxTheoKm** [comment]
Die beiden betr. Titel hatte ich tatsächlich schon korrigiert. Im Moment finde ich nur Titel mit überzähligen Leerzeichen, bei denen davor noch ein Anführungszeichen steht:
491868383
492169020
